### ISSUE TITLE
Ability to delete multiple database layers in a single operation

### ISSUE THREAD

**Issue by timlinux** [qgis/QGIS#39095]
**Feature description.**

Currently when I select more than one layer in the browser in the PostGIS browser (I guess it is the same for other data sources) I get prompted for each layer individually. Could we add a 'yes to all' so that all the layers can be deleted in one operation?

![deleting-multiple-tables-in-qgis](https://user-images.githubusercontent.com/178003/94668540-14e36280-0308-11eb-833d-c1b35d22b150.gif)

CC @elpaso

**Comment by uclaros** [comment]
Preferably with a message stating the total number of tables / views that are about to be dropped.

**Comment by elpaso** [comment]
I agree that it would be nice to have, I also though about it in the past.

Unfortunately this is not trivial and requires an API change (or a new method), the logic that actually deletes the layer is buried inside the provider's dataitem `QgsDataItemGuiProvider` (or the legacy `QgsLayerItem`) subclass and the confirmation dialog as well as all other checks are performed inside the `deleteLayer` method of these classes. You can select layers from multiple providers and click on "Delete selected layers..." and the individual provider `QgsDataItemGuiProvider` implementations are called. 

We probably need to add an additional argument to `QgsDataItemGuiProvider::deleteLayer(bool force = false)` to delete the layers no questions asked and to move the confirmation to the calling method (which is `QgsLayerItemGuiProvider::deleteLayers`).

Side note: this feature was not part of the grant proposal 2020 but has been there for years.

**Comment by nirvn** [comment]
@elpaso , one possible quick and dirty short term implementation: [x] do not ask for confirmation for the next minute :)

**Comment by timlinux** [comment]
> Side note: this feature was not part of the grant proposal 2020 but has been there for years.

Yes acknowledged thanks - I have run into the issue in the past but never got to writing a feature request. I CC'd you because this kind of stuff seems to be on your radar :-) In my case I have thousands of temporary views I want to delete, probably better done with a bit of SQL anyway but I figured I would try in QGIS....

**Comment by elpaso** [comment]
Yeah, the browser is definitely on my radar!  I'll see what I can do.

**Comment by uclaros** [comment]
> one possible quick and dirty short term implementation: [x] do not ask for confirmation for the next minute :)

A safer quick and dirty hack would be to have the focus default to the *Yes* button. That way you can spam space to get past the confirmation dialogs!

**Comment by rduivenvoorde** [comment]
What about, currently the 'View deleted succesfully' is a popup you have to click away, but if we make this a message in the messagebar? That takes away one click? :-)

**Comment by nyalldawson** [comment]
@rduivenvoorde we should definitely do that!

**Comment by elpaso** [comment]
> What about, currently the 'View deleted succesfully' is a popup you have to click away, but if we make this a message in the messagebar? That takes away one click? :-)

I must have missed that one, because I've recently been through most of the notifications and moved to the new `notify()` function that does exactly that: https://github.com/qgis/QGIS/pull/39006/files

**Comment by elpaso** [comment]
@rduivenvoorde @nyalldawson actually I didn't miss that one: https://github.com/qgis/QGIS/pull/39006/files#diff-5a28d4100aa4174bdbc911376ee16be7R135

**Comment by uclaros** [comment]
Nice! That means that now one can spam *Y* to have the equivalent of *yes to all*.
However, selecting *No* misleadingly notifies that `Delete Layer : Item Layer xxx cannot be deleted.`
